File: frontend/ProcessExplorer-ui/src/types/ProcInfo.ts
export interface ProcInfo {
    pid: number;
    parentPid: number;
    name: string;
    cpu: number;
    workingSetKb: number;
    privateBytesKb: number;
    threadCount: number;
    handleCount: number;
    diskKbPerSec: number;
    startTimeUnixMs: number;
    executablePath: string | null;
}

File: backend/ProcessExplorer.Api/Features/ProcessMonitoring/ProcInfo.cs
namespace ProcessExplorer.Api.Features.ProcessMonitoring;

public record ProcInfo
(
    int Pid,
    int ParentPid,
    string Name,
    double Cpu,
    long WorkingSetKb,
    long PrivateBytesKb,
    int ThreadCount,
    int HandleCount,
    double DiskKbPerSec,
    long StartTimeUnixMs,
    string? ExecutablePath
);
